data/sql_setup/prepare_schedule_a.sql

```diff
@@ -92,27 +323,18 @@ begin
         -- visit here:
         -- https://www.postgresql.org/docs/current/static/plpgsql-statements.html#PLPGSQL-STATEMENTS-DIAGNOSTICS
         if FOUND then
-            two_year_transaction_period = cast(get_cycle(view_row.rpt_yr) as smallint);
-
-            if two_year_transaction_period >= start_year then
+            if view_row.election_cycle >= start_year then
                 delete from ofec_sched_a_queue_old where sub_id = view_row.sub_id;
-                insert into ofec_sched_a_queue_old values (view_row.*, timestamp, two_year_transaction_period);
+                insert into ofec_sched_a_queue_old values (view_row.*, current_timestamp, view_row.election_cycle);
+                IF tg_op = 'DELETE' THEN
+                    DELETE FROM ofec_sched_a_master WHERE sub_id = view_row.sub_id;
+                END IF;
             end if;
         end if;
 
+    if tg_op = 'DELETE' then
         return old;
     elsif tg_op = 'UPDATE' then
-        select into view_row * from fec_fitem_sched_a_vw where sub_id = old.sub_id;
-
-        if FOUND then
-            two_year_transaction_period = cast(get_cycle(view_row.rpt_yr) as smallint);
-
-            if two_year_transaction_period >= start_year then
-                delete from ofec_sched_a_queue_old where sub_id = view_row.sub_id;
-                insert into ofec_sched_a_queue_old values (view_row.*, timestamp, two_year_transaction_period);
-            end if;
-        end if;
-
```

Comment:
🤔 I wonder if we even need the `UPDATE` portion in the delete trigger anymore... if we're going to process these sans-queues and take care of them in real time, then the `AFTER` trigger that is setup for `INSERT` and `UPDATE` should suffice since we have to wait until a record is (hopefully) available through the view, right?  In which case, we don't need to worry about tracking the changes ourselves anymore.

Reply:
Looks like this is resolved now with this code being removed, right?

Reply:
We need the `UPDATE` portion for the aggregates.